experimental_design/assay_simulator.py

```diff
@@ -0,0 +1,279 @@
+
+        # Select the parameters for the noise based on where the fluorescence was taken
+        if self.geometry == 'top':
+            self.sigma = np.mean(pymc_data['sigma_top'][0][t_equil:])
+        else:
+            self.sigma = np.mean(pymc_data['sigma_bottom'][0][t_equil:])
+
+    def simulate_fluorescence(self, DeltaG=None, p_total=None, noisy=True):
+        """
+        Predict the fluorescence of the complex using AssayTools posterior for a given protein concentration
+        and range of ligand concentrations.
+
+        Parameters
+        ----------
+        DeltaG: float
+            The binding free energy in thermal units.
+        p_total: float
+            Concentration of protein in M.
+        noisy: bool
+            Whether to add detector noise to the predicted fluorescence.
+
+        Returns
+        --------
+        Fmodel: numpy.ndarray
+            The expected fluorescence given the supplied model parameters
+        """
+        if p_total is None:
+            p_total = self.p_total
+        if DeltaG is None:
+            DeltaG = self.DeltaG
+
+        # Predict the concentrations of the complex, free protein and free ligand
+        [P_free, L_free, PL] = TwoComponentBindingModel.equilibrium_concentrations(DeltaG, p_total, self.l_total)
+
+        # Model the fluorescence
+        if self.inner_filter:
+            Fmodel = self.IF_i * (self.F_PL * PL + self.F_L * L_free + self.F_P * P_free + self.F_buffer * self.path_length) + self.IF_i_plate * self.F_plate
+            if noisy:
+                Fmodel += np.random.normal(loc=0.0, scale=self.sigma, size=len(self.l_total))
+        else:
+            Fmodel = self.F_PL * PL + self.F_L * L_free + self.F_P * P_free + self.F_buffer * self.path_length
+            if noisy:
+                Fmodel += np.random.normal(loc=0.0, scale=self.sigma, size=len(self.l_total))
+
+        return Fmodel
+
+    def fit_deltaG(self, p_total=None):
+        """
+        Estimate the binding free energy given the fluorescence model parameters using least-squares fitting.
+
+        Parameters
+        ----------
+        p_total: float
+            Concentration of protein in M.
+
+        Returns
+        -------
+        fit: float
+            The estimated binding free energy in thermal units
+        """
+
+        if p_total is None:
+            p_total = self.p_total
+
+        # The fluorescence data that will be fit to
+        target = self.simulate_fluorescence(p_total)
+
+        def sum_of_squares(DeltaG, target=target):
+            """
+            The sum of squares between model fluorescence and the target
+            """
+            model = self.simulate_fluorescence(DeltaG, p_total, noisy=False)
+            return np.sum((model - target)**2)
+
+        # Start the initial guess within about 10% of the "true" value
+        guess = self.DeltaG + np.random.normal(loc=0, scale=0.1 * np.abs(self.DeltaG))
```

Comment:
It looks like this is the only source of uncertainty currently being added to your model---the initial starting guess for the least-squares fit---since `noisy=False` in `sum_of_squares` above. Is that right?